include/zephyr/irq.h

```diff
@@ -22,6 +24,11 @@
 extern "C" {
 #endif
 
+#if defined(CONFIG_MULTI_LEVEL_INTERRUPTS) && CONFIG_MAX_IRQ_PER_AGGREGATOR > 0
+BUILD_ASSERT((LOG2(CONFIG_MAX_IRQ_PER_AGGREGATOR) + 1) <= BIT64(CONFIG_1ST_LEVEL_INTERRUPT_BITS),
+	"CONFIG_MAX_IRQ_PER_AGGREGATOR is too large");
+#endif
```

Comment:
I believe that this could be used to show how the existing script fails for some targets currently, but it might not be useful if the 3 bitmasks are non-uniform in size.

Reply:
I think this has a logic error. The check I was trying to implement was to ensure that the total bits needed  to represent the MAX_IRQ_PER_AGGREGATOR would fit inside the number of first level bits. Otherwise this would cut into the second level bits. Essentially causing the same problem we have now internally.

Reply:
I think we should continue supporting that. I was thinking of this as an addition to use in the event it is impossible to use the current hierarchy pattern as opposed to replacing it entirely.